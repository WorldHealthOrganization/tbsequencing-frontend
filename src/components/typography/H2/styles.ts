import { css } from '@emotion/react';
import { font, fontSizes } from '../fontSizes';

export const h2Styles = css({
  fontFamily: font.bold,
  fontSize: fontSizes.h2,
  lineHeight: '28px',
  fontWeight: 'bold',
});
