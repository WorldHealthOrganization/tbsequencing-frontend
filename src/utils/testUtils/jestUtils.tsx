import React, { ReactElement } from 'react';
import { render as rtlRender } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { reducer } from '../../app/store';

const getWrapper = (
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
  } = {},
) => ({ children }: { children: ReactElement }) => (
  <Provider store={store}>{children}</Provider>
);

function render(
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
    ...renderOptions
  } = {},
) {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const getTreeJSONWithRedux = (ui: ReactElement, {
  // @ts-ignore
  preloadedState,
  store = configureStore({ reducer, preloadedState }),
} = {}) => {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  const tree = renderer
    .create(<Wrapper>{ui}</Wrapper>)
    .toJSON();

  return tree;
};

export const wrappedInRouter = (
  component: ReactElement,
) => <BrowserRouter>{component}</BrowserRouter>;

export * from '@testing-library/react';
export { render, getTreeJSONWithRedux };
