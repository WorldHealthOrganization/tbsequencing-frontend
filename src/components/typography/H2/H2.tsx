/** @jsxImportSource @emotion/react */
import React, { ReactNode } from 'react';
import { css, SerializedStyles } from '@emotion/react';
import * as styles from './styles';
import BaseText from '../BaseText';

interface Props {
  children: ReactNode;
  style?: SerializedStyles;
}

export const H2 = ({ children, style }: Props) => {
  const injectedStyles = style || css({});

  return (
    <BaseText
      style={[styles.h2Styles, injectedStyles]}
    >
      {children}
    </BaseText>
  );
};

export default H2;
