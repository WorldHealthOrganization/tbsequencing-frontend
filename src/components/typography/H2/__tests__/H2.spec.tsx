import React from 'react';
import { getTreeJSONWithRedux, render, screen } from '../../../../utils/testUtils/jestUtils';
import H2 from '../H2';

describe('H2 component test', () => {
  const mockText = 'mockText';

  test('renders H2 component with a proper text', () => {
    render(
      <H2>{mockText}</H2>,
    );

    const foundElement = screen.queryByText(mockText);

    expect(foundElement).toBeTruthy();
  });

  it('Should render correctly', () => {
    const tree = getTreeJSONWithRedux(
      <H2>{mockText}</H2>,
    );

    expect(tree).toMatchSnapshot();
  });
});
