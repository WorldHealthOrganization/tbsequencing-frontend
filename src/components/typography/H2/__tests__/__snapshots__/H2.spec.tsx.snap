// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H2 component test Should render correctly 1`] = `
<p
  className="MuiTypography-root MuiTypography-body1 css-nwtw2h-MuiTypography-root-BaseText"
>
  mockText
</p>
`;
